src/main/java/com/squareup/javapoet/CodeBlock.java

```diff
@@ -102,6 +104,53 @@ public static CodeBlock of(String format, Object... args) {
     return new Builder().add(format, args).build();
   }
 
+  /**
+   * Joins {@code codeBlocks} into a single {@link CodeBlock}, each separated by {@code separator}.
+   * For example, joining {@code String s}, {@code Object o} and {@code int i} using {@code ", "}
+   * would produce {@code String s, Object o, int i}.
+   */
+  public static CodeBlock join(Iterable<CodeBlock> codeBlocks, String separator) {
+    Collection<CodeBlock> collection;
+    if (codeBlocks instanceof Collection) {
+      collection = (Collection<CodeBlock>) codeBlocks;
+    } else {
+      collection = new ArrayList<>();
+      codeBlocks.forEach(collection::add);
+    }
+    return collection.stream().collect(joining(separator));
+  }
+
+  /**
+   * A {@link Collector} implementation that joins {@link CodeBlock} instances together into one
+   * separated by {@code separator}. For example, joining {@code String s}, {@code Object o} and
+   * {@code int i} using {@code ", "} would produce {@code String s, Object o, int i}.
+   */
+  public static Collector<CodeBlock, ?, CodeBlock> joining(String separator) {
+    return Collector.of(
+        () -> new CodeBlockJoiner(separator, builder()),
+        CodeBlockJoiner::add,
+        CodeBlockJoiner::merge,
+        CodeBlockJoiner::join);
+  }
+
+  /**
+   * A {@link Collector} implementation that joins {@link CodeBlock} instances together into one
+   * separated by {@code separator}. For example, joining {@code String s}, {@code Object o} and
+   * {@code int i} using {@code ", "} would produce {@code String s, Object o, int i}.
+   */
+  public static Collector<CodeBlock, ?, CodeBlock> joining(
+      String separator, String prefix, String suffix) {
+    Builder builder = builder().add("$N", prefix);
```

Comment:
Why `$N` here? Should it rather take `CodeBlock`s instead?

Reply:
I figured that passing a string (with a literal) is likely easier, and `$N` maintains the exact string.

I thought about having code blocks. I think that's probably less common (at that point, you might just want to add the joined code block to a `CodeBlock.Builder`?)

Reply:
I like this. Avoids escaping weirdness too.